src/engine/MaxConstraint.h

```diff
@@ -262,10 +246,28 @@ class MaxConstraint : public PiecewiseLinearConstraint
                  : static_cast<unsigned>( phase ) - MAX_VARIABLE_TO_PHASE_OFFSET;
     }
 
+    /*
+      Eliminate the case corresponding to the given input variable to Max.
+    */
+    void eliminateCase( unsigned variable );
+
     /*
       Return true iff f or the elements are not all within bounds.
     */
     bool haveOutOfBoundVariables() const;
+
+    /*
+     * Returns a boolean value indicating if at least one input variable was eliminated (True)
+     * or not (False)
+     */
+    bool wereVariablesEliminated() const;
+
+    /*
+      For DEBUG only:
+      Return true if all the element in _elements are feasible
+      based on their bounds.
+    */
+    void checkAllInputElementsFeasible() const;
```

Comment:
Since its used only for debugging, maybe use assert rather then check to make that distinction? 

Is AllInputElements in the function name redundant? While the max variable might be an element, by itself it can never be infeasible, so making that distinction may be unnecessary. What do you think?

Reply:
Removed this method since I didn't end up using it.